nix/supervisord-cluster/clusterlib.py

```diff
@@ -1,167 +1,369 @@
+        #    change_address = self.genesis_utxo_addr
 
         # TODO: calculate from current tip
-        ttl=100000
-        # Estimate Fee
-        #fee = self.estimateFee(ttl=ttl, txins=len(txins), txouts=len(txouts), signing_keys = [ self.genesisUtxoSkey ] + signing_keys)
+        ttl = 100000
+
         # TODO: unhardcode genesis utxo
-        change = (self.genesisUtxoAddr, 0)
-        txouts.append(change)
-        total_input_amount = 0
-        # TODO: use a fold to be less hacky
-        for txin in txins:
-            total_input_amount += txin[2]
+        change = (self.genesis_utxo_addr, 0)
+        txouts_copy.append(change)
 
         deposit_amount = 0
-
-
-        txins_combined = list(map( lambda x: f"{x[0]}#{x[1]}", txins))
-        txouts_combined = list(map( lambda x: f"{x[0]}+{x[1]}", txouts))
-
-        txin_args = self.prependFlag("--tx-in", txins_combined)
-
-        txout_args = self.prependFlag("--tx-out", txouts_combined)
-
-        cert_args = self.prependFlag("--certificate-file", certificates)
-
-        build_args = [ "--ttl", str(ttl), "--fee", "0", "--out-file", "tx.body" ] + txin_args + txout_args + cert_args
-
-        if proposal:
-            build_args.extend(["--update-proposal-file", proposal])
-
-
-        # Build TX
-
-        self.CLI(["cardano-cli", "shelley", "transaction", "build-raw"] + build_args )
-        # Estimate Fee
-        fee = self.estimateFee("tx.body", txins=len(txins), txouts=len(txouts), witnesses=len(signing_keys))
-
-        txouts[-1] = (self.genesisUtxoAddr, (total_input_amount - fee - deposit_amount))
-        txouts_combined = list(map( lambda x: f"{x[0]}+{x[1]}", txouts))
-        txout_args = self.prependFlag("--tx-out", txouts_combined)
-        build_args = [ "--ttl", str(ttl), "--fee", str(fee), "--out-file", "tx.body" ] + txin_args + txout_args + cert_args
-
-        if proposal:
-            build_args.extend(["--update-proposal-file", proposal])
-
-        self.CLI(["cardano-cli", "shelley", "transaction", "build-raw"] + build_args )
-
-        if proposal:
-            build_args.extend(["--update-proposal-file", proposal])
-
-
-        # Sign TXw
-
-        key_args = self.prependFlag("--signing-key-file", signing_keys)
-
-        sign_args = [ "--tx-body-file", "tx.body", "--out-file", "tx.signed", "--testnet-magic", str(self.networkMagic) ] + key_args
-        self.CLI(["cardano-cli", "shelley", "transaction", "sign"] + sign_args )
-
-        # Submit TX
-        try:
-            self.CLI(["cardano-cli", "shelley", "transaction", "submit", "--testnet-magic", str(self.networkMagic), "--tx-file", "tx.signed"])
-        except CLIError as e:
-            print("Failed to submit transaction!")
-            print(f"txins: {txins} txouts: {txouts} signing keys: {signing_keys}")
-            raise e
-
-    def getAddress(self, payment=None, stake=None):
-        args = []
-        if not payment and stake:
-            raise CLIError("Must set payment, stake or both")
```

Comment:
Seems like this was a logic bug. 

`not payment and stake` would have only been `True` in the event that `bool(payment) == False` and `bool(stake) == True`. Therefore this doesn't catch the case where neither `payment` nor `stake` have been provided.